airflow/models.py

```diff
@@ -346,10 +346,10 @@ def kill_zombies(self, session=None):
 
                     # now set non db backed vars on ti
                     ti.task = task
-                    ti.test_mode = False
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
```

Comment:
@msumit It should definitely not come the global config. Please explain why it would always be False here (you could be right, didn't trace it throughout the code).